File: code/backend/src/domain/entities/Hospede.ts
export class Hospede {
    private id: string;
    private nome: string;
    private sobrenome: string;
    private cpf: string;
    private email: string;

    constructor(id: string, nome: string, sobrenome: string, cpf: string, email: string) {
        this.id = id;
        this.nome = nome;
        this.sobrenome = sobrenome;
        this.cpf = cpf;
        this.email = email;
        
        this.validar();
    }

    private validar(): void {
        if (!this.nome || this.nome.trim().length === 0) {
            throw new Error("Nome é obrigatório.");
        }
        if (!this.cpf) {
            throw new Error("CPF é obrigatório.");
        }
        if (!this.email || !this.email.includes('@')) {
            throw new Error("Email inválido.");
        }
    }

    public getId(): string {
        return this.id;
    }

    public getNomeCompleto(): string {
        return `${this.nome} ${this.sobrenome}`;
    }

    public getCpf(): string {
        return this.cpf;
    }

    public getEmail(): string {
        return this.email;
    }
}

File: code/stack-2/backend/src/main/java/com/hotel/model/Hospede.java
package com.hotel.model;

/**
 * Entidade de Domínio Hóspede.
 * Representa o cliente do hotel.
 * Decisão: CPF é utilizado como identificador único natural nas regras de negócio.
 */
public class Hospede {
    private String nome;
    private String sobrenome;
    private String cpf;
    private String email;

    public Hospede() {
        // Necessário para desserialização
    }

    public Hospede(String nome, String sobrenome, String cpf, String email) {
        this.nome = nome;
        this.sobrenome = sobrenome;
        this.cpf = cpf;
        this.email = email;
    }

    // Getters e Setters
    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }

    public String getSobrenome() { return sobrenome; }
    public void setSobrenome(String sobrenome) { this.sobrenome = sobrenome; }

    public String getCpf() { return cpf; }
    public void setCpf(String cpf) { this.cpf = cpf; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
}
